ExtractUserDataActivity/handler.ts

```diff
@@ -463,27 +470,56 @@ export const queryAllUserData = (
       })
     ),
     // step 3: queries notifications statuses
-    TE.chain(
+    TE.bind("notificationStatuses", ({ notifications }) =>
+      findAllNotificationStatuses(notificationStatusModel, notifications)
+    ),
+    // step 4: queries profile service preferences
+    TE.bindW("servicesPreferences", () =>
+      pipe(
+        TE.tryCatch(
+          async () => servicePreferencesModel.findAllByFiscalCode(fiscalCode),
+          () =>
+            ActivityResultQueryFailure.encode({
+              kind: "QUERY_FAILURE",
+              reason: "Error while searching the profile service preferences"
+            })
+        ),
+        TE.map(flattenAsyncIterator),
+        TE.map(asyncIteratorToArray),
+        TE.chain(
+          (promise: Promise<ReadonlyArray<t.Validation<ServicePreference>>>) =>
+            TE.tryCatch(
+              () => promise,
+              () =>
+                ActivityResultQueryFailure.encode({
+                  kind: "QUERY_FAILURE",
+                  reason: "Error with the async operator"
+                })
+            )
+        ),
+        TE.map(ROA.rights)
+      )
+    ),
+    TE.map(
       ({
         profile,
+        notifications,
         messages,
         messagesView,
         messageContents,
         messageStatuses,
-        notifications
-      }) =>
-        sequenceS(TE.ApplicativePar)({
```

Comment:
this sequenceS only contained one task so i replaced with a bind for a more convenient structure

Reply:
Could be more conveniente, but is the execution order always parallel?

Reply:
as you can see there is only one taskeither in the function (`notificationStatuses`), so parallelism doesn't matter here

Reply:
...or use apS insteand of bind https://gcanti.github.io/fp-ts/guides/do-notation.html#examples